extensions/resource-deployment/src/ui/notebookWizard/notebookWizardToolsAndEulaPage.ts

```diff
@@ -0,0 +1,516 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { EOL } from 'os';
+import * as nls from 'vscode-nls';
+import { AgreementInfo, DeploymentProvider, DeploymentProviderBase, ITool, ResourceType, ToolStatus } from '../../interfaces';
+import { createFlexContainer } from '../modelViewUtils';
+import * as loc from '../../localizedConstants';
+import { IToolsService } from '../../services/toolsService';
+import { getErrorMessage } from '../../common/utils';
+import { WizardPageBase } from '../wizardPageBase';
+import { WizardBase } from '../wizardBase';
+import { Model } from '../model';
+
+
+const localize = nls.loadMessageBundle();
+
+
+export class ToolsAndEulaPage<W extends WizardBase<WizardPageBase<W, M>, M>, M extends Model> extends WizardPageBase<W, M> {
+	private form!: azdata.FormBuilder;
+	private view!: azdata.ModelView;
+	private toolRefreshTimestamp: number = 0;
+	private _optionsContainer!: azdata.FlexContainer;
+	private _toolsTable!: azdata.TableComponent;
+	private _optionDropDownMap: Map<string, azdata.DropDownComponent> = new Map();
+	private _toolsLoadingComponent!: azdata.LoadingComponent;
+	private _agreementContainer!: azdata.DivContainer;
+	private _agreementCheckboxChecked: boolean = false;
+	private _installToolButton!: azdata.ButtonComponent;
+	private _installationInProgress: boolean = false;
+	private _tools: ITool[] = [];
+	private _eulaValidationSucceeded: boolean = false;
+	private _isInitialized = false;
+	private _isDoneButtonEnabled = false;
+
+
+	public set agreementCheckboxChecked(value: boolean) {
+		this._agreementCheckboxChecked = value;
+	}
+
+	public get agreementCheckboxChecked(): boolean {
+		return this._agreementCheckboxChecked;
+	}
+
+	public get resourceType(): ResourceType {
+		return this.wizard.resourceType;
+	}
+
+	public set resourceProvider(provider: DeploymentProviderBase) {
+		this.wizard.resourceProvider = provider;
+	}
+
+	public get toolsService(): IToolsService {
+		return this.wizard.toolsService;
+	}
+
+	constructor(wizard: W, _pageIndex: number = 0) {
+		super('', localize('notebookWizard.toolsAndEulaPageTitle', "Deployment Pre-Requisite"), wizard);
```

Comment:
No title?

Reply:
Deployment pre -req should be the title. I will fix it.